System/Vector.cpp

```diff
@@ -35,9 +35,11 @@ namespace RTE {
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	Vector & Vector::SetMagnitude(float newMag) {
-		Vector temp(*this);
-		SetXY(newMag, 0);
-		AbsRotateTo(temp);
+		if (IsZero()) {
+			SetXY(newMag, 0.0F);
+		} else {
+			*this *= newMag / GetMagnitude();
```

Comment:
This is pretty minor but I would put a bracket around `newMag / GetMagnitude` just for safety of order of operations.
I'm pretty sure the *= should be evaluated last, but I don't memorize order of operations for these sorts of things so I'm not 100% of that, and if it's not that'd mean you're doing (this * newMag) / GetMagnitude, which is more operations cause Vector * float has to do 2 mults.